Changes on modbus.json page and Modbus page (#11)
intelie/gitbook-drilling: collector/protocols/modbus.md

# MODBUS

{% hint style="info" %}

> This capability is provided by **Liverig Collector 2.28.0** or higher&#x20;

Reviewer: ```suggestion
This capability is provided by **LiveRig Collector 2.28.0** or higher
```

{% endhint %}

> The MODBUS protocol is capable of receiving data from a PLC (programmable logic controllers) device via TCP connections if Liverig is older than 5.0.0, otherwise the data can be received  via

Reviewer: ```suggestion
The MODBUS protocol is capable of receiving data from a PLC (programmable logic controllers) device via TCP connections if LiveRig is older than 5.0.0, otherwise the data can be received  via
```

TCP, UDP and Serial connections.

> There are a number of memory areas defined in the MODBUS specification. Here, are the types of data supported, by the Liverig Collector:

Reviewer: ```suggestion
There are a number of memory areas defined in the MODBUS specification. Here, are the types of data supported, by the LiveRig Collector:
```

  * Input Register - Short input value, usually representing an analog input to the PLC
  * Holding Register - Short value, usually representing an analog output from the PLC

## Configuring the `endpoint` field for Modbus sources

> ### For Liverig versions 5.0.0 or above

Reviewer: ```suggestion
### LiveRig Collector 5.0.0 or above
```

> If you are using a Liverig version 5.0.0 or above, you can change the connection protocol by declaring the endpoint using the

Reviewer: ```suggestion
If you are using a LiveRig version 5.0.0 or above, you can change the connection protocol by declaring the endpoint using the
```

format: `{code}:{transport}://{ip-address}:{port}?{options}`. Here is a list of options:

> - Code: `modbus-tcp`*(default)*, `modbus-adu`, `modbus-ascii`;
> - Transport: `tcp`*(default)*, `udp`, `serial`;
> - Options: `request-timeout`, `unit-identifier` (This will use the following
>   structure: `?{option-1-name}={option-1-value}&{option-2-name}={option-2-value}`)

Reviewer: ```suggestion
- **code**: `modbus-tcp`*(default)*, `modbus-adu`, `modbus-ascii`;
- **transport**: `tcp`*(default)*, `udp`, `serial`;
- **options**: `request-timeout`, `unit-identifier` (This will use the following structure: `?{option-1-name}={option-1-value}&{option-2-name}={option-2-value}`)
```

> The `Transport`, `Port` and `Options` fields are optional. The default port value is `502` and if the `Code` value is

Reviewer: ```suggestion
The `transport`, `port` and `options` fields are optional. The default port value is `502` and if the `code` value is
```

just `modbus`, it will default to the `modbus-tcp` value.

**Example TCP Modbus endpoint:**

`modbus-tcp:tcp//127.0.0.1:1552`

**Example Serial Modbus endpoint:**
- For Linux Systems:
  `modbus-adu:serial:/dev/ttyUSB0`
- For Windows Systems:
  `modbus-adu:serial:COM1`

> **OBS:** *You need to add a special permission to access serial Sources on Linux. This can be done by adding the user to

Reviewer: ```suggestion
**Note:** *You need to add a special permission to access serial Sources on Linux. This can be done by adding the user to
```

the `dialout` group using a `sudo usermod -a -G dialout $USER` command.*

> ### For liverig versions older than 5.0.0

Reviewer: ```suggestion
### LiveRig Collector previous than 5.0.0
```

> If you are using a Liverig version below 5.0.0, you cannot change the connection protocol, so the endpoint format must be `modbus://{ip-address}:{port}`

Reviewer: ```suggestion
If you are using a LiveRig 4.x or bellow, you cannot change the connection protocol, so the endpoint format must be `modbus://{ip-address}:{port}`
```
